### PR TITLE
Universal deposit fee

### PR DESCRIPTION
**Pull request by EdNoepel**
## Description

Replace the unutilized deposit fee, which charged 24 hours of interest, with a deposit fee which applies to all buckets, and charges 8 hours of interest.  In addition to `addQuoteToken`, the fee shall be charged when calling `moveQuoteToken` when moving liquidity to a lower price.

## Purpose

Resolve Kirill audit issue M-05 (https://github.com/k1rill-fedoseev/audits/blob/master/solo/Ajna.md):
> Lenders may avoid the unutilized deposit fee by manipulating book with collateral.
> Book may be manipulated using a small loan to avoid the unutilized deposit fee.

## Impact

Gas improvement expected for `addQuoteToken` and `moveQuoteToken`, as the LUP only needs to be calculated once instead of twice.  Interface changes on both methods to remove the parameter letting caller control whether TX should revert if LUP moved above their target bucket.

Pool contract sizes are down; ERC721Pool is down to 24,075B (97.96%).

## Tasks

- [ ] Changes to protocol contracts are covered by unit tests executed by CI.
- [x] Protocol contract size limits have not been exceeded.
- [ ] Gas consumption for impacted transactions have been compared with the target branch, and nontrivial changes cited in the _Impact_ section above.
- [ ] Scope labels have been assigned as appropriate.
- [ ] Invariant tests have been manually executed as appropriate for the nature of the change.

### CODE REVIEW

**Review by ith-harvey — COMMENTED** [review]
looking

**Review by ith-harvey — APPROVED** [review]
src changes look solid. Wondering if we should also create a separate unit test for this?

File: src/libraries/external/LenderActions.sol [hunk 1]
```diff
@@ -285,11 +273,8 @@ library LenderActions {
             })
         );
 
-        lup_ = Deposits.getLup(deposits_, poolState_.debt);
-        // apply unutilized deposit fee if quote token is moved from above the LUP to below the LUP
-        if (vars.fromBucketPrice >= lup_ && vars.toBucketPrice < lup_) {
-            if (params_.revertIfBelowLup) revert PriceBelowLUP();
-
+        // apply deposit fee if moving to a lower-priced bucket
+        if (params_.fromIndex < params_.toIndex) {
```
Comment (MikeHathaway) [inline src/libraries/external/LenderActions.sol:277]: Should the deposit fee be charged on any move?

**Review by MikeHathaway — APPROVED** [review]
LGTM. Much cleaner without revert param and additional branches.

File: src/libraries/external/LenderActions.sol (hunk 1 above)
Comment (EdNoepel) [reply src/libraries/external/LenderActions.sol:277]: We just changed this today; fee will only be charged when moving to a lower price.

**Review by mattcushman — APPROVED** [review]
lgtm

**Review by prat-gpt — APPROVED** [review]
LGTM

**Pull request merged**